pwiz_tools/Skyline/FileUI/PeptideSearch/ImportFastaControl.cs

```diff
@@ -420,72 +420,92 @@ public bool ImportFasta(IrtStandard irtStandard)
                 // FASTA, set FragmentType='p' and AutoSelect=true
                 var docCurrent = DocumentContainer.Document;
                 var docNew = ImportPeptideSearch.PrepareImportFasta(docCurrent);
-
                 var nodeInsert = _sequenceTree.SelectedNode as SrmTreeNode;
                 IdentityPath selectedPath = nodeInsert != null ? nodeInsert.Path : null;
                 var newPeptideGroups = new List<PeptideGroupDocNode>();
 
-                if (!_fastaFile)
+                // if user provided FASTA content, write it to a temporary file
+                string fastaFilepath = _fastaFile ? tbxFasta.Text : Path.GetTempFileName();
+                try
                 {
-                    FastaText = tbxFasta.Text;

```

Comment:
@nickshulman Here's a case where a using declaration would come in pretty handy to avoid this Path.GetTempFileName()/try/finally construct which incidentally makes the diff hard to read. If I made a TemporaryFile class like the existing TemporaryDirectory class, I could use that with a using declaration instead of this try/finally. (The using statement version would probably still make a hard-to-read diff like this one due to indentation changes.)

Reply:
I am not sure I understand why a temporary file needs to be created here. Do we need to pass that file path to some other code that really requires a real file on disk and could not take a stream?